lib/std/heap.zig

```diff
@@ -8,6 +8,129 @@ const c = std.c;
 const Allocator = std.mem.Allocator;
 const windows = std.os.windows;
 
+const page_size_darwin: ?comptime_int = if (builtin.os.tag.isDarwin())
+    switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be, .aarch64_32 => 16 << 10,
+        else => null,
+    }
+else
+    null;
+
+// -- <https://devblogs.microsoft.com/oldnewthing/20210510-00/?p=105200>
+const page_size_windows: ?comptime_int = if (builtin.os.tag == .windows and builtin.os.version_range.windows.min.isAtLeast(.xp))
+blk: {
+    break :blk switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // SuperH => 4 << 10,
+        .mips, .mipsel, .mips64, .mips64el => 4 << 10,
+        .powerpc, .powerpcle, .powerpc64, .powerpc64le => 4 << 10,
+        // DEC Alpha => 8 << 10,
+        // Itanium => 8 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be, .aarch64_32 => 4 << 10,
+        else => null,
+    };
+} else null;
+
+const page_size_os: ?comptime_int = page_size_darwin orelse page_size_windows;
+
+/// A compile time known upper bound on page size.
+pub const page_size_cap: usize = page_size_os orelse switch (builtin.cpu.arch) {
+    // Common knowledge.
+    .wasm32, .wasm64 => 64 << 10,
+    .x86, .x86_64 => 4 << 10,
+    .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_32, .aarch64_be => 64 << 10,
+    // Explicitly only 4kb.
+    // https://refspecs.linuxbase.org/ELF/zSeries/lzsabi0_zSeries.html#AEN798
+    .s390x => 4 << 10,
+    // Source: Linux mips/Kconfig.
+    .mips, .mipsel, .mips64, .mips64el, .loongarch32, .loongarch64 => 64 << 10,
+    // Source: csky/Kconfig only selects HAVE_PAGE_SIZE_4KB.
+    .csky => 4 << 10,
+    // Source: Hexagon's page.h in Linux accepts CONFIG_PAGE_SIZE_1MB.
+    .hexagon => 1024 << 10,
+    // Source: Zig's own libc page.h for arc.
+    .arc => 16 << 10,
+    // Source: Wikipedia "Page (computer memory)"
+    .powerpc, .powerpc64, .powerpc64le, .powerpcle => 64 << 10,
+    .riscv32, .riscv64 => 4 << 10,
+    .sparc, .sparcel => 256 << 10,
+    .sparc64 => 64 << 10,
+    // Rare architectures with little support.
+    else => @compileError("Does pageSize() apply to this architecture? If so, file an issue."),
+};
+
+/// Compile time known minimum page size.
+pub const page_size: usize = page_size_os orelse switch (builtin.cpu.arch) {
```

Comment:
I think we should name this `min_page_size`, and `page_size_cap` should be `max_page_size`. As they are, it's not immediately obvious what either of the names mean. ("cap" could be confused to have to do with capabilities or something.)